mne/channels/tests/test_standard_montage.py

```diff
@@ -73,3 +78,59 @@ def test_standard_superset():
     for key, value in m_1020._get_ch_pos().items():
         if key not in ('O10', 'O9'):
             assert_allclose(c_1005[key], value, atol=1e-4, err_msg=key)
+
+
+def _simulate_artinis_octamon():
+    """Simulate artinis OctaMon channel data from numpy data.
+
+    This is to test data that is imported with missing or incorrect montage
+    info. This data can then be used to test the set_montage function.
+    """
+    data = np.random.normal(size=(16, 100))
+    ch_names = ['S1_D1 hbo', 'S1_D1 hbr', 'S2_D1 hbo', 'S2_D1 hbr',
+                'S3_D1 hbo', 'S3_D1 hbr', 'S4_D1 hbo', 'S4_D1 hbr',
+                'S5_D2 hbo', 'S5_D2 hbr', 'S6_D2 hbo', 'S6_D2 hbr',
+                'S7_D2 hbo', 'S7_D2 hbr', 'S8_D2 hbo', 'S8_D2 hbr']
+    ch_types = ['hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr']
+    sfreq = 10.  # Hz
+    info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+    raw = RawArray(data, info, verbose=True)
+
+    return raw
+
+
+def test_set_montage_artinis():
```

Comment:
A few additional tests I would like to see:
* What happens if there is less channels in the data than locations in the montage?
* What happens if the channel names don't correspond to the montage?
* What happens if there are more channels in the data than in the montage?
All these things should throw a meaningful error. I think most of this should be picked up by the existing set_montage code, but as fNIRS acts quite differently we should check that reasonable error text is thrown.

Reply:
Also we need to test that the set_montage function operates correctly on amplitude, od, and hb data.